### ISSUE TITLE
Backstage 1.6.0 breaks Azure DevOps plugin

### ISSUE THREAD

**Issue by LvffY** [backstage/backstage#13880]
## Expected Behavior

Be able to migrate the Azure DevOps plugin to backstage 1.6.0.

## Actual Behavior

Passing to backstage 1.6.0 breaks the Azure DevOps plugin.

When running our apps (that previously worked in backstage 1.3.0), I have the following error 

![image](https://user-images.githubusercontent.com/37327656/192748499-bb94214d-a7ba-437f-bd46-844a469ddad4.png).

I see in my logs that my backend is running and correctly fetch the azure catalogs so I think this is related to the react-router migration but not sure about that.

## Steps to Reproduce

1. Upgrade to last backstage version.
2. run `yarn dev`

## Your Environment

- Browser Information: 

- Output of `yarn backstage-cli info`: 

```text
OS:   Linux 5.10.102.1-microsoft-standard-WSL2 - linux/x64
node: v16.15.1
yarn: 1.22.19
cli:  0.20.0-next.0 (installed)
backstage:  1.6.0

Dependencies:
  @backstage/app-defaults                               1.0.6
  @backstage/backend-common                             0.13.5, 0.15.1
  @backstage/backend-plugin-api                         0.1.2
  @backstage/backend-tasks                              0.3.5
  @backstage/catalog-client                             1.1.0
  @backstage/catalog-model                              1.1.1
  @backstage/cli-common                                 0.1.10
  @backstage/cli                                        0.19.0
  @backstage/config-loader                              1.1.4
  @backstage/config                                     1.0.2
  @backstage/core-app-api                               1.1.0
  @backstage/core-components                            0.11.1, 0.9.5
  @backstage/core-plugin-api                            1.0.6
  @backstage/dev-utils                                  1.0.6
  @backstage/errors                                     1.1.1
  @backstage/integration-react                          1.1.4
  @backstage/integration                                1.3.1
  @backstage/plugin-api-docs                            0.8.9
  @backstage/plugin-app-backend                         0.3.36
  @backstage/plugin-auth-backend                        0.16.0
  @backstage/plugin-auth-node                           0.2.5
  @backstage/plugin-azure-devops-backend                0.3.15
  @backstage/plugin-azure-devops-common                 0.3.0
  @backstage/plugin-azure-devops                        0.2.0
  @backstage/plugin-catalog-backend-module-azure        0.1.7
  @backstage/plugin-catalog-backend-module-gitlab       0.1.7
  @backstage/plugin-catalog-backend-module-msgraph      0.4.2
  @backstage/plugin-catalog-backend                     1.4.0
  @backstage/plugin-catalog-common                      1.0.6
  @backstage/plugin-catalog-graph                       0.2.21
  @backstage/plugin-catalog-import                      0.8.12
  @backstage/plugin-catalog-node                        1.1.0
  @backstage/plugin-catalog-react                       1.1.4
  @backstage/plugin-catalog                             1.5.1
  @backstage/plugin-explore-react                       0.0.21
  @backstage/plugin-explore                             0.3.40
  @backstage/plugin-github-actions                      0.5.9
  @backstage/plugin-home                                0.4.25
  @backstage/plugin-kubernetes-backend                  0.7.2
  @backstage/plugin-kubernetes-common                   0.4.2
  @backstage/plugin-kubernetes                          0.7.2
  @backstage/plugin-org                                 0.5.9
  @backstage/plugin-permission-common                   0.6.4
  @backstage/plugin-permission-node                     0.6.5
  @backstage/plugin-permission-react                    0.4.5
  @backstage/plugin-proxy-backend                       0.2.30
  @backstage/plugin-scaffolder-backend                  1.6.0
  @backstage/plugin-scaffolder-common                   1.2.0
  @backstage/plugin-scaffolder                          1.6.0
  @backstage/plugin-search-backend-module-pg            0.4.0
  @backstage/plugin-search-backend-node                 1.0.2
  @backstage/plugin-search-backend                      1.0.2
  @backstage/plugin-search-common                       1.0.1
  @backstage/plugin-search-react                        1.1.0
  @backstage/plugin-search                              1.0.2
  @backstage/plugin-sonarqube-backend                   0.1.1
  @backstage/plugin-sonarqube                           0.4.1
  @backstage/plugin-stack-overflow                      0.1.5
  @backstage/plugin-tech-insights-backend-module-jsonfc 0.1.20
  @backstage/plugin-tech-insights-backend               0.5.2
  @backstage/plugin-tech-insights-common                0.2.6
  @backstage/plugin-tech-insights-node                  0.3.4
  @backstage/plugin-tech-insights                       0.3.0
  @backstage/plugin-tech-radar                          0.5.16
  @backstage/plugin-techdocs-backend                    1.3.0
  @backstage/plugin-techdocs-module-addons-contrib      1.0.4
  @backstage/plugin-techdocs-node                       1.4.0
  @backstage/plugin-techdocs-react                      1.0.4
  @backstage/plugin-techdocs                            1.3.2
  @backstage/plugin-todo-backend                        0.1.33
  @backstage/plugin-todo                                0.2.11
  @backstage/plugin-user-settings                       0.4.8
  @backstage/release-manifests                          0.0.6
  @backstage/test-utils                                 1.2.0
  @backstage/theme                                      0.2.16
  @backstage/types                                      1.0.0
  @backstage/version-bridge                             1.0.1
```

**Comment by Rugvip** [comment]
I'm not able to reproduce this in the example app. What does the installation of the extension look like? It should be something like this:

```tsx
<EntityLayout.Route path="/ci-cd" title="CI/CD">
  {cicdContent}
</EntityLayout.Route>
```

**Comment by awanlin** [comment]
Just wanted to comment that this is working fine for us without any issues on 1.6.0 - we only use the Builds (CI/CD) and Pull Request views though. But I would suspect that the other features should be fine too. We also have migrated to React Router 6.0 as per the docs: https://backstage.io/docs/tutorials/react-router-stable-migration

**Comment by LvffY** [comment]
@Rugvip @awanlin Thanks for your answers :)

After some digging, I found the error came from the **overview** content. To be explicit, I had the following **EntityPage** : 

```node
const cicdContent = (
    <EntitySwitch>
    // ... 
        <EntitySwitch.Case if={isAzureDevOpsAvailable}>
            <Grid item md={6}>
                <EntityAzurePipelinesContent defaultLimit={25}/>
                <EntityAzurePullRequestsContent defaultLimit={25}/>
            </Grid>
        </EntitySwitch.Case>
        <EntitySwitch.Case if={isAzurePipelinesAvailable}>
            <EntityAzurePipelinesContent defaultLimit={25} />
        </EntitySwitch.Case>
    </EntitySwitch>
);

const entityWarningContent = (
    <>
        <EntitySwitch>
            <EntitySwitch.Case if={isOrphan}>
                <Grid item xs={12}>
                    <EntityOrphanWarning />
                </Grid>
            </EntitySwitch.Case>
        </EntitySwitch>

        <EntitySwitch>
            <EntitySwitch.Case if={hasCatalogProcessingErrors}>
                <Grid item xs={12}>
                    <EntityProcessingErrorsPanel />
                </Grid>
            </EntitySwitch.Case>
        </EntitySwitch>
    </>
);

const overviewContent = (
    <Grid container spacing={3} alignItems="stretch">
        // ...
        <EntitySwitch>
            <EntitySwitch.Case if={isAzureDevOpsAvailable}>
                <Grid item md={6}>*/}
                    <EntityAzurePipelinesContent />
                    <EntityAzurePullRequestsContent />
                </Grid>*/}
            </EntitySwitch.Case>
        </EntitySwitch>
        // ...
    </Grid>
);

const serviceEntityPage = (
    <EntityLayout>
        <EntityLayout.Route path="/" title="Overview">
            {overviewContent}
        </EntityLayout.Route>

        // ...

        <EntityLayout.Route
            if={isAzureDevOpsAvailable}
            path="/pull-requests"
            title="Pull Requests">
            <EntityAzurePullRequestsContent defaultLimit={25} />
        </EntityLayout.Route>
        // ...
    </EntityLayout>
);

const libraryEntityPage = (
    <EntityLayout>
        <EntityLayout.Route path="/" title="Overview">
            {overviewContent}
        </EntityLayout.Route>

        // ...

        <EntityLayout.Route
            if={isAzureDevOpsAvailable}
            path="/pull-requests"
            title="Pull Requests">
            <EntityAzurePullRequestsContent defaultLimit={25}/>
        </EntityLayout.Route>

        //...
    </EntityLayout>
);


const websiteEntityPage = (
    <EntityLayout>
        <EntityLayout.Route path="/" title="Overview">
            {overviewContent}
        </EntityLayout.Route>

        // ...
    </EntityLayout>
);

/**
 * NOTE: This page is designed to work on small screens such as mobile devices.
 * This is based on Material UI Grid. If breakpoints are used, each grid item must set the `xs` prop to a column size or to `true`,
 * since this does not default. If no breakpoints are used, the items will equitably share the available space.
 * https://material-ui.com/components/grid/#basic-grid.
 */

const defaultEntityPage = (
    <EntityLayout>
        <EntityLayout.Route path="/" title="Overview">
            {overviewContent}
        </EntityLayout.Route>

        // ...
    </EntityLayout>
);

const componentPage = (
    <EntitySwitch>
        <EntitySwitch.Case if={isComponentType('service')}>
            {serviceEntityPage}
        </EntitySwitch.Case>

        <EntitySwitch.Case if={isComponentType('website')}>
            {websiteEntityPage}
        </EntitySwitch.Case>
        {/* Ensure libraries will have the same page as services */}
        <EntitySwitch.Case if={isComponentType('library')}>
            {libraryEntityPage}
        </EntitySwitch.Case>

        <EntitySwitch.Case>{defaultEntityPage}</EntitySwitch.Case>
    </EntitySwitch>
);

// ...
```
Just by commenting the part in the **overviewContent** I made my backstage work again. Is there anything I'm missing ? 

cc @Neemys

**Comment by Rugvip** [comment]
@LvffY thank you for the thorough report! Turns our this was a bug in our routing logic, fixed over in #13989 🎉
